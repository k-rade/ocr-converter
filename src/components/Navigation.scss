.convert-options {
    position: relative;
    display: flex;
    align-self: flex-start;
    overflow: hidden;
    width: 100%;
    border-bottom: 1px solid var(--color-gray-700);
}

.convert-option {
    border: none;
    outline: none;
    background-color: transparent;
    font-family: var(--font);
    font-size: 0.875rem;
    font-weight: 600;
    color: var(--color-gray-100);
    padding: 1rem;
    display: flex;
    align-items: center;
    justify-content: center;
    gap: 0.75rem;
    text-align: center;
    cursor: pointer;
    position: relative;
    border-bottom: 2px solid transparent;

    &:disabled {
        cursor: not-allowed;
    }
}

.active.convert-option {
    color: var(--color-white);
    border-bottom: 2px solid var(--color-primary-600);
}