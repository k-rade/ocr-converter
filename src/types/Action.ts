export enum ActionType {
  SET_IMAGE_FILE,
  SET_IMAGE_URL,
  SET_LANGUAGE,
  SET_PROGRESS,
  SET_ERROR,
  CONVERT_IMAGE_INIT,
  CONVERT_IMAGE_SUCCESS,
  CONVERT_IMAGE_ERROR,
  INIT_NEW_CONVERSION,
}

export interface Action {
  type: ActionType;
  payload?: any;
}
