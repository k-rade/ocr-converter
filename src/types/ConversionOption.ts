export enum ConversionOption {
  FILE_UPLOAD,
  BY_LINK,
}
