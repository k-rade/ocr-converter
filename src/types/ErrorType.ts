export enum ErrorType {
  NO_ERROR,
  CONVERSION_FAILED,
  UNSUPPORTED_FILE_FORMAT,
  MAX_FILE_SIZE_EXCEEDED,
}
