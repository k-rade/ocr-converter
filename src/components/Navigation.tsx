import { ConversionOption } from "../types/ConversionOption";
import { ReactComponent as FileTextIcon } from "../icons/file-text.svg";
import { ReactComponent as LinkIcon } from "../icons/link.svg";

import "./Navigation.scss";

interface Props {
  isLoading: boolean;
  conversionOption: ConversionOption;
  changeConversionOption: (conversionOption: ConversionOption) => void;
}

const Navigation = ({
  isLoading,
  conversionOption,
  changeConversionOption,
}: Props): JSX.Element => {
  return (
    <div className="convert-options">
      <button
        disabled={isLoading}
        className={`convert-option ${
          conversionOption === ConversionOption.FILE_UPLOAD ? "active" : ""
        }`}
        onClick={() => changeConversionOption(ConversionOption.FILE_UPLOAD)}
      >
        File Upload
      </button>
      <button
        disabled={isLoading}
        className={`convert-option ${
          conversionOption === ConversionOption.BY_LINK ? "active" : ""
        }`}
        onClick={() => changeConversionOption(ConversionOption.BY_LINK)}
      >
        By Link
      </button>
    </div>
  );
};

export default Navigation;
