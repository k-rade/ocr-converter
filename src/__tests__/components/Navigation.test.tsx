import { fireEvent, render } from "@testing-library/react";
import Navigation from "../../components/Navigation";
import { ConversionOption } from "../../types/ConversionOption";

describe("Navigation", (): void => {
  it("should render", (): void => {
    const { baseElement } = render(
      <Navigation
        isLoading={false}
        conversionOption={ConversionOption.FILE_UPLOAD}
        changeConversionOption={jest.fn}
      />
    );

    expect(baseElement).toMatchSnapshot();
  });

  it("should display disabled buttons", (): void => {
    const { getByText } = render(
      <Navigation
        isLoading={true}
        conversionOption={ConversionOption.FILE_UPLOAD}
        changeConversionOption={jest.fn}
      />
    );

    expect(getByText("File Upload")).toHaveAttribute("disabled");
    expect(getByText("By Link")).toHaveAttribute("disabled");
  });

  it.each([
    [ConversionOption.FILE_UPLOAD, "File Upload"],
    [ConversionOption.BY_LINK, "By Link"],
  ])(
    "should change conversion option to %s when clicked on %s",
    (option: ConversionOption, link: string): void => {
      const mockChangeConversionOption = jest.fn();
      const { getByText } = render(
        <Navigation
          isLoading={false}
          conversionOption={ConversionOption.FILE_UPLOAD}
          changeConversionOption={mockChangeConversionOption}
        />
      );

      fireEvent.click(getByText(link));

      expect(mockChangeConversionOption).toHaveBeenCalledWith(option);
    }
  );
});
