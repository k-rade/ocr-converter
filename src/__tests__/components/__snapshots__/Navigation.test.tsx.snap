// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation should render 1`] = `
<body>
  <div>
    <div
      class="convert-options"
    >
      <button
        class="convert-option active"
      >
        File Upload
      </button>
      <button
        class="convert-option "
      >
        By Link
      </button>
    </div>
  </div>
</body>
`;
